independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationsTransformer.java

```diff
@@ -64,4 +83,202 @@ default boolean isMethod() {
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must contain ANY of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        public Builder whenContainsAny(List<DotName> annotationNames) {
+            return when(context -> Annotations.containsAny(context.getAnnotations(), annotationNames));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must contain ANY of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        public Builder whenContainsAny(DotName... annotationNames) {
+            return whenContainsAny(List.of(annotationNames));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must contain ANY of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        @SafeVarargs
+        public final Builder whenContainsAny(Class<? extends Annotation>... annotationNames) {
+            return whenContainsAny(
+                    Arrays.stream(annotationNames).map(a -> DotName.createSimple(a.getName())).collect(Collectors.toList()));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must NOT contain any of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        public Builder whenContainsNone(List<DotName> annotationNames) {
+            return when(context -> !Annotations.containsAny(context.getAnnotations(), annotationNames));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must NOT contain any of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        public Builder whenContainsNone(DotName... annotationNames) {
+            return whenContainsNone(List.of(annotationNames));
+        }
+
+        /**
+         * {@link TransformationContext#getAnnotations()} must NOT contain any of the given annotations.
+         * 
+         * @param annotationNames
+         * @return self
+         */
+        @SafeVarargs
+        public final Builder whenContainsNone(Class<? extends Annotation>... annotationNames) {
+            return whenContainsNone(
+                    Arrays.stream(annotationNames).map(a -> DotName.createSimple(a.getName())).collect(Collectors.toList()));
+        }
+
+        /**
+         * The transformation logic is only performed if the given predicate is evaluated to true. Multiple predicates are
+         * logically-ANDed.
+         * 
+         * @param predicate
+         * @return self
+         */
+        public Builder when(Predicate<TransformationContext> predicate) {
+            if (predicate == null) {
+                this.predicate = predicate;
+            } else {
+                this.predicate = predicate.and(predicate);
```

Comment:
Nice subtle typo :-) Must be

```suggestion
                this.predicate = this.predicate.and(predicate);
```

Reply:
Good catch! There was actually one more error - `if (predicate == null)` => `if (this.predicate == null)` so I changed the argument name to `when` and fixed the test too. Thanks!

Reply:
Haha correct! Thanks!